packages/app/server/impl.ts

```diff
@@ -85,13 +87,17 @@ export async function spawnServer({ configuration, root }: ApplicationData): Pro
       // cf https://github.com/mscdex/httpolyglot/issues/3#issuecomment-173680155
       // HTTPS:
       if (data.readUInt8(0) === 22) {
-        socket = await tlsManager.process(socket);
-        setConnectionProtocol(socket, 'https');
+        const tlsSocket = await tlsManager.process(socket, ['http/1.1']);
+        handleSocket(tlsSocket);
       } else {
         await Promise.resolve();
-        setConnectionProtocol(socket, 'http');
+        let protocol = 'http';
+        if (socket instanceof TLSSocket) {
+          protocol = 'https';
+        }
+        setConnectionProtocol(socket, protocol);
```

Comment:
I would have use the ternary operator directly (and remove the protocol variable).

Reply:
Thank you @fbasso! I have changed the code to take your comment into account!